### ISSUE TITLE
Problems with PayPal and quantity casting

### ISSUE THREAD

**Issue by mschadegg** [woocommerce/woocommerce#12554]
The #12293 has caused problems for several shops.

When you cast quantity as int, valuable information is lost, if you have shops, where you use decimal in quantity.

I know PayPal not support decimal in quantity, but it has been possible to overcome this, by converting the quantity, title and price in the filter 'woocommerce_paypal_args'. This is no longer possible without creating the PayPal args from scratch again, because information about the quantity is lost.

Here is how I use the filter, but now information about quantity is lost.

```
public function paypal_args($args, $order)
    {
        foreach ($args as $key => $arg) {
            if (preg_match('/quantity_/', $key)) {
                $number = substr($key, 9);
                if (!is_int($args['quantity_' . $number])) {
                    $args['amount_' . $number] = round($args['amount_' . $number] * $args['quantity_' . $number], 2);
                    $args['item_name_' . $number] = $args['item_name_' . $number] . ' x ' . $args['quantity_' . $number];
                    $args['quantity_' . $number] = 1;
                }
            }
        }
        return $args;
    }
```
I understand why you cast it as int, but can you create a filter, so it is possible to manipulate the information, or implement the fix above?

```
$line_item = apply_filters(
	'woocommerce_paypal_line_item', 
	array(
		'item_name_' . $index => html_entity_decode( wc_trim_string( $item_name ? $item_name : __( 'Item', 'woocommerce' ), 127 ), ENT_NOQUOTES, 'UTF-8' ),
		'quantity_' . $index => (int) $quantity,
		'amount_' . $index => (float) $amount,
		'item_number_' . $index => $item_number,
	), 
	$index, 
	$item_name, 
	$quantity, 
	$amount, 
	$item_number
);
	
$this->line_items = array_merge($this->line_items, $line_item);
```

**Comment by mikejolley** [comment]
Why don't you just try to enable the filter which sends the order as 1 big item instead?

**Comment by ammist** [comment]
Please note that this filter is not available in WooCommerce 2.6.14, only 2.7 and up.

**Comment by awkoenig** [comment]
I've ran into this issue on version 3.2.1.  Client is using decimal quantities.  We've tried both options: do not send line items to pay paypal and add another line.  It is using the paypal express.  Do I need to manually implement the filter?
